### ISSUE TITLE
Process not exiting with process.exit() when running a busy isolate

### ISSUE THREAD

**Issue by kiddkai** [laverdet/isolated-vm#260]
Hi again! Thank you for the amazing module! We are running into an issue that creates dangling node.js processes. Here's how I can reproduce it in a simple example:

I also tested: Node12 and Node16.6, they shared the same issue. For now, I have to call `process.kill(pid)` to exit the processes.

```js
const { pid } = require("process");
const ivm = require("isolated-vm");
const isolate = new ivm.Isolate({ memoryLimit: 128 });

let i = 0;
const run = async () => {
  const context = isolate.createContextSync();

  const jail = context.global;

  jail.setSync("global", jail.derefInto());

  jail.setSync("log", function (...args) {
      console.log(pid, i++, ...args)
  });

  const script = await isolate.compileScript(`
        while (true) {
            log('hello');
        }
    `);

  setTimeout(() => {
      console.log('exit pricess', pid);
      isolate.dispose();
      process.exit(0);
     console.log('exit pricess finished', pid);
  }, 1000)

  script.run(context, {
      release: true, 
      timeout: 1000
  });
};

run()
```

From my understanding, the expected behavior is the process exit itself. But now, it just stuck at the message:

```
exit process [pid]
```

And here's the screenshot:

![image](https://user-images.githubusercontent.com/1309053/134452934-80535e47-40a1-4e36-9786-0e0bb4e75067.png)

**Comment by laverdet** [comment]
Yeah this is a known issue. If you wait long enough it should conceivably quit. What's happened here is the scheduler got filled up with a bunch of tasks from the infinite loop. Those need to be flushed before it's able to dispose. I definitely need a solution in place because this condition is pretty bad.

**Comment by laverdet** [comment]
I recommend throwing fantasies of a clean exit code out the window. v8 just isn't built for this. Open up your Chrome Devtools right now, drop in `Array(1e9).fill('a').join('');`, and watch your tab die. I opened a ticket 4 years ago to the v8 team about this issue and it is unfixed to this day. Resilience in this manner is just not a design goal of v8. Your runner infrastructure should be robust to out of control processes and know how to dispatch a SIGKILL when needed.
